saml-core-api/src/main/java/org/keycloak/dom/saml/v2/assertion/AttributeValueElementType.java

```diff
@@ -0,0 +1,27 @@
+package org.keycloak.dom.saml.v2.assertion;
+
+import java.io.Serializable;
+
+import org.w3c.dom.Element;
+
+public class AttributeValueElementType implements Serializable {
+
+    private String type;
```

Comment:
The `type` should be a `QName` to include namespace definition, to allow for schema validation (see `xs` or `ns` in the following example):
```xml
        <saml:AttributeValue xsi:type="xs:string">value</saml:AttributeValue>
        <saml:AttributeValue xsi:type="ns:Role" xmlns:ns="urn:my:resource:type">value</saml:AttributeValue>
```
In the case of serialization of the type, if the namespace part of `QName` is defined, in needs to be declared in the `AttributeValue` attribute. If it is not, it would allow declarations from the PR like:
```xml
        <saml:AttributeValue xsi:type="Role">value</saml:AttributeValue>
```
Yet the namespace-less variants do not allow schema validation so should be not used.

Reply:
I implemented the changes as suggested.

Reply:
I just read the specification one more time and I think you were right with the first comment: The type attribute at attributeValue is obsolete, when the child is an element with its own type and namespace declaration.
The type is only necessary at simple elements like string and date, but not at complex elements.

So I removed the type and the AttributeValueType completly.

Sorry for the confusion.